modules/hostap/src/supp_api.c

```diff
@@ -1408,6 +1410,75 @@ int hapd_config_network(struct hostapd_iface *iface,
 }
 #endif
 
+int supplicant_ap_status(const struct device *dev, struct wifi_iface_status *status)
+{
+	int ret = 0;
+	struct hostapd_iface *iface = NULL;
+	struct hostapd_config *conf = NULL;
+	struct hostapd_data *hapd = NULL;
+	struct hostapd_bss_config *bss = NULL;
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+	iface = get_hostapd_handle(dev);
+	if (!iface) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Interface %s not found", dev->name);
+		goto out;
+	}
+	conf = iface->conf;
+	if (!conf) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Conf %s not found", dev->name);
+		goto out;
+	}
+	bss = conf->bss[0];
+	if (!bss) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Bss_conf %s not found", dev->name);
+		goto out;
+	}
+	hapd = iface->bss[0];
+	if (!hapd) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Bss %s not found", dev->name);
+		goto out;
+	}
+
+	int proto;
+	int key_mgmt;
```

Comment:
Please introduce variables at the start of the function

Reply:
modified